src/lewis/adapters/stream.py

```diff
@@ -37,27 +37,63 @@ class StreamHandler(asynchat.async_chat):
     def __init__(self, sock, target, stream_server):
         asynchat.async_chat.__init__(self, sock=sock)
         self.set_terminator(b(target.in_terminator))
-        self.target = target
-        self.buffer = []
+        self._readtimeout = target.readtimeout
+        self._readtimer = 0
+        self._target = target
+        self._buffer = []
 
         self._stream_server = stream_server
 
         self._set_logging_context(target)
         self.log.info('Client connected from %s:%s', *sock.getpeername())
 
+    def process(self, msec):
+        if not self._buffer:
+            return
+
+        if self._readtimer >= self._readtimeout and self._readtimeout != 0:
```

Comment:
By the way, PyCharm highlights this line and tells me to "Simplify chained comparison"...

This is usually a hint to chain conditions like  like `if a > b and c < b:` into `if a > b > c:`.

But I can't figure out what it wants from me here. How do you chain `>=` and `!= 0`? 
`0 < self._readtimeout <= self._readtimer`?
That's not quite the same, and not nearly as clear in intention. Any ideas? Am I missing something or is PyCharm just being crazy here?